src/app/visit/visit.hpp

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include <memory>
+#include <set>
+#include <string>
+
+class Doctor;
+
+class Visit : public std::enable_shared_from_this<Visit>
+{
+    Visit();
+    static std::set<Visit *> visitExtent;
+    std::shared_ptr<Doctor> docAssociation_;
+
+    std::string visitInfo_;
+
+  public:
+    ~Visit();
+    static std::shared_ptr<Visit> createVisit(std::shared_ptr<Doctor> doc);
+    static std::set<Visit *> getExtent();
```

Comment:
I think, that we should not provide access to whole extent, but have getters returning concrete visit (or more visitis) for specified arguments, lets say (Doctor) -> all visits for doctor (Room)-> all visits for room, (Patient) -> all visits for Patient.
(Date/Hour) -> all visits happening at the same time.
(Doctor, Patient) -> filtered accordingly and so on.

Otherwise we will  have to filter all Visits in the extent in many places in the code manually (redundancy etc.)